### PR TITLE
Saves newscaster data and photos to round logs folder

### PR DESCRIPTION
**Pull request by Jordie0608**
Makes newscaster save their channels, stories, comments and wanted issue along with associated photos at the end of the round for @nfreader (although tbh I'm not sure he asked for it). Formatting is not removed from the text.

Photos are named upon attachment as a shortened md5 of `world.time` + `rand(1,99)` to avoid collisions and saved to current log folder as pngs.

Example json of what this all looks like:
~~~
{
  "1": {
    "channel name": "Station Announcements",
    "author": "SS13",
    "censored": 0,
    "author censored": 0,
    "messages": {
      "author": "Central Command",
      "time stamp": "12:01:41",
      "censored": 0,
      "author censored": 0,
      "photo file": "",
      "photo caption": "",
      "body": "Security Report<br><br>Thanks to the tireless efforts of our security and intelligence divisions, there are currently no credible threats to Festive Hazard Four. All station construction projects have been authorized. Have a secure shift!",
      "comments": []
    }
  },
  "2": {
    "channel name": "somechannel",
    "author": "Hunter Zaun (Captain)",
    "censored": 0,
    "author censored": 0,
    "messages": {
      "author": "Hunter Zaun (Captain)",
      "time stamp": "12:00:28",
      "censored": 0,
      "author censored": 0,
      "photo file": "b4989",
      "photo caption": "goose",
      "body": "<font face=\"Verdana\">quack</font>",
      "comments": []
    }
  },
  "3": {
    "channel name": "anotherchannel",
    "author": "Captain (Captain)",
    "censored": 1,
    "author censored": 1,
    "messages": {
      "author": "Captain (Captain)",
      "time stamp": "12:01:20",
      "censored": 1,
      "author censored": 0,
      "photo file": "692f9",
      "photo caption": "",
      "body": "<font face=\"Verdana\">boringmessage</font>",
      "comments": []
    }
  },
  "wanted": {
    "author": "Captain (Captain)",
    "criminal": "bananaman one",
    "description": "fruit are banned",
    "photo file": "c7c75"
  }
}
~~~

### CODE REVIEW

File: code/__HELPERS/roundend.dm [hunk 1]
```diff
@@ -117,7 +137,7 @@
 	//Collects persistence features
 	if(mode.allow_persistence_save)
 		SSpersistence.CollectData()
-
+	gather_newscaster()
```
Comment (Jordie0608) [inline code/__HELPERS/roundend.dm]: move to `gather_roundend_feedback()`

File: code/game/machinery/newscaster.dm [hunk 2]
```diff
@@ -157,7 +161,11 @@ GLOBAL_LIST_EMPTY(allCasters)
 	for(var/obj/machinery/newscaster/NEWSCASTER in GLOB.allCasters)
 		NEWSCASTER.update_icon()
 
-
+/datum/newscaster/feed_network/proc/save_photo(icon/photo)
+	var/photo_file = copytext(md5("[world.time][rand(1,99)]"), 1, 6)
+	var/icon/p = icon(photo, frame = 1)
+	fcopy(p, "[GLOB.log_directory]/photos/[photo_file].png")
```
Comment (Jordie0608) [inline code/game/machinery/newscaster.dm]: md5 photo images and reuse matches

**Review by Jordie0608 — COMMENTED** [review]
\>pull request authors can't request changes on their own pull requests

**Comment by nfreader** [comment]
Yay!! This should be a lot of fun going forward. I’m assuming ‘messages’ is an array, if a given newscast has multiple posts?

**Comment by Jordie0608** [comment]
Yes, same for comments, but note there's only ever one wanted.

**Pull request merged**